### ISSUE TITLE
hexo 3.2.2 not have `hexo server` ?

### ISSUE THREAD

**Issue by sumaolin** [hexojs/hexo#2088]
## Environment Info

Node version

```
v4.4.5
```

Your site `_config.yml`

```
# Hexo Configuration
# Hexo Configuration
## Docs: https://hexo.io/docs/configuration.html
## Source: https://github.com/hexojs/hexo/

# Site
title: 苏茂林
subtitle: Kevin.Su's Blog
description: 苏茂林, 个人博客, Kevin.Su's Blog, 关于前端技术的的杂记！
author: Kevin Su
language: zh-Hans
timezone:

# URL
## If your site is put in a subdirectory, set url as 'http://yoursite.com/child' and root as '/child/'
url: http://sumaolin.com/
root: /
permalink: :year/:month/:day/:title/
permalink_defaults:

# Directory
source_dir: source
public_dir: public
tag_dir: tags
archive_dir: archives
category_dir: categories
code_dir: downloads/code
i18n_dir: :lang
skip_render:

# Writing
new_post_name: :title.md # File name of new posts
default_layout: post
titlecase: false # Transform title into titlecase
external_link: true # Open external links in new tab
filename_case: 0
render_drafts: false
post_asset_folder: false
relative_link: false
future: true
highlight:
  enable: true
  line_number: true
  auto_detect: false
  tab_replace:

# Category & Tag
default_category: uncategorized
category_map:
tag_map:

# Date / Time format
## Hexo uses Moment.js to parse and display date
## You can customize the date format as defined in
## http://momentjs.com/docs/#/displaying/format/
date_format: YYYY-MM-DD
time_format: HH:mm:ss

# Pagination
## Set per_page to 0 to disable pagination
per_page: 10
pagination_dir: page

# Extensions
## Plugins: https://hexo.io/plugins/
plugins: 
  - hexo-qiniu-sync
qiniu:
  offline: false
  sync: true
  bucket: sumaolin
  # access_key: hpuB89Wsfc2l8zgx-6A5-P5xI-l0cYWE19KpPdX-
  # secret_key: Hd1XwcNuwwNsN0ADxjGrIDk6INcY_EEuMUXMBzvb
  secret_file: ./sec/qn.json
  dirPrefix: static
  urlPrefix: http://sumaolin.reekly.com/static
  up_host: http://upload.qiniu.com
  local_dir: static
  update_exist: true
  image: 
    folder: img
    extend: 
  js:
    folder: js
  css:
    folder: css


## Themes: https://hexo.io/themes/
theme: next
avatar: http://tp4.sinaimg.cn/1765384767/180/5749265416/1
duoshuo_shortname: sumaolin
baidu_analytics: 154101e86ea67b8b420e04051c6e0b2a


# Deployment
## Docs: https://hexo.io/docs/deployment.html
deploy:
  type: git
  repo:
    github: git@github.com:sumaolin/sumaolin.github.com.git,master
    coding: git@git.coding.net:sumaolin/sumaolin.git,coding-pages


server:
  port: 4000
  log: false
  ip: 127.0.0.1
  compress: false
  header: true
...
```

Your theme `_config.yml`

```
# when running hexo in a subdirectory (e.g. domain.tld/blog), remove leading slashes ( "/archives" -> "archives" )
menu:
  home: /
  #categories: /categories
  #about: /about
  archives: /archives
  tags: /tags
  #commonweal: /404.html


# Enable/Disable menu icons.
# Icon Mapping:
#   Map a menu item to a specific FontAwesome icon name.
#   Key is the name of menu item and value is the name of FontAwsome icon.
#   When an question mask icon presenting up means that the item has no mapping icon.
menu_icons:
  enable: true
  # Icon Mapping.
  home: home
  about: user
  categories: th
  tags: tags
  archives: archive
  commonweal: heartbeat


# Place your favicon.ico to /source directory.
favicon: ./favicon.ico

# Set default keywords (Use a comma to separate)
keywords: "Hexo,next"

# Set rss to false to disable feed link.
# Leave rss as empty to use site's feed link.
# Set rss to specific value if you have burned your feed already.
rss:

# Custom Logo.
# !!Only available for Default Scheme currently.
# Options:
#   enabled: [true/false] - Replace with specific image
#   image: url-of-image   - Images's url
custom_logo:
  enabled: false
  image:



# Code Highlight theme
# Available value: normal | night | night eighties | night blue | night bright
# https://github.com/chriskempson/tomorrow-theme
highlight_theme: night blue


# MathJax Support
mathjax:


# Schemes
scheme: Mist


# Sidebar, available value:
#  - post    expand on posts automatically. Default.
#  - always  expand for all pages automatically
#  - hide    expand only when click on the sidebar toggle icon.
#  - remove  Totally remove sidebar including sidebar toggle icon.
sidebar: post
#sidebar: always
#sidebar: hide
#sidebar: remove


# Automatically scroll page to section which is under <!-- more --> mark.
scroll_to_more: true


# TOC in the Sidebar
toc:
  enable: true

  # Automatically add list number to toc.
  number: true


# Automatically Excerpt
auto_excerpt:
  enable: false
  length: 150

# Use Lato font
use_font_lato: true

# Make duoshuo show UA
# user_id must NOT be null when admin_enable is true!
# you can visit http://dev.duoshuo.com get duoshuo user id.
duoshuo_info:
  ua_enable: true
  admin_enable: false
  user_id: 0
  duoshuo_hotartical: true
  #admin_nickname: ROOT
duoshuo_share: true
baidu_analytics: 154101e86ea67b8b420e04051c6e0b2a
google_analytics: UA-52231954-2

# Facebook SDK Support.
# https://github.com/iissnan/hexo-theme-next/pull/410
facebook_sdk:
  enable: false
  app_id:       #<app_id>
  fb_admin:     #<user_id>
  like_button:  #true
  webmaster:    #true


## DO NOT EDIT THE FOLLOWING SETTINGS
## UNLESS YOU KNOW WHAT YOU ARE DOING


# Use velocity to animate everything.
use_motion: true

# Fancybox
fancybox: true

# Static files
vendors: vendors
css: css
js: js
images: images

# Theme version
version: 0.4.5.2

...
```

Plugin version

```
hexo-site@0.0.0 E:\hexoBlog\hexo\hexo
+-- hexo@3.2.2
+-- hexo-deployer-git@0.1.0
+-- hexo-generator-archive@0.1.4
+-- hexo-generator-category@0.1.3
+-- hexo-generator-feed@1.2.0
+-- hexo-generator-index@0.2.0
+-- hexo-generator-sitemap@1.1.2
+-- hexo-generator-tag@0.2.0
+-- hexo-qiniu-sync@1.4.5
+-- hexo-renderer-ejs@0.1.1
+-- hexo-renderer-marked@0.2.11
+-- hexo-renderer-stylus@0.3.1
`-- hexo-server@0.1.3
```
## For BUG
- `hexo server` not work
- update hexo to V3.2.2
- hexo --help 

``` bash
λ hexo --help
INFO  -----------------------------------------------------------
INFO  qiniu state: online
INFO  qiniu sync:  true
INFO  qiniu local dir:  static
INFO  qiniu url:   http://sumaolin.reekly.com/static
INFO  -----------------------------------------------------------
Usage: hexo <command>

Commands:
  clean     Removed generated files and cache.
  config    Get or set configurations.
  deploy    Deploy your website.
  generate  Generate static files.
  help      Get help on a command.
  init      Create a new Hexo folder.
  list      List the information of the site
  migrate   Migrate your site from other system to Hexo.
  new       Create a new post.
  publish   Moves a draft post from _drafts to _posts folder.
  qiniu     Qiniu sync
  render    Render files with renderer plugins.
  version   Display version information.

Global Options:
  --config  Specify config file instead of using _config.yml
  --cwd     Specify the CWD
  --debug   Display all verbose messages in the terminal
  --draft   Display draft posts
  --safe    Disable all plugins and scripts
  --silent  Hide output on console

For more help, you can use 'hexo help [command]' for the detailed information
or you can check the docs: http://hexo.io/docs/
```

there is not 'hexo server' command

**Comment by mizzao** [comment]
`hexo server` is now separate from `hexo-cli`: `npm install -g hexo-server`.

**Comment by longdivision** [comment]
I had a similar problem to @sumaolin after updating from Hexo 3.1 to 3.2 where `hexo server` would not run and `server` was not listed as a command when running `hexo help`.

For me the issue was caused by the following in my `_config.yml` (as advised by the plugin docs):

```
plugins:
  - hexo-generator-cname
```

I was able to resolve the issue by removing the plugin from my `_config.yml`. All plugins loaded and worked as expected (including hexo-generator-cname).

I think what's happening here is that the `plugins` block in `_config.yml` overrides the automatic loading of plugins from node_modules, so it looked like I only wanted to load one plugin. This explains why hexo-server wasn't working, as it is now a plugin and would not have been loaded. The code for this can be seen [here](https://github.com/hexojs/hexo/blob/5234c4a85dc6cd418e9a1c169e43de169cf98e95/lib/hexo/load_plugins.js#L18).

It looks like the plugins block is also used to load plugins from a local plugins folder (see https://github.com/hexojs/hexo/commit/f6407dc946c2a686731b2480ee6ed0b37f54c5f1), so maybe it makes sense to load both the plugins specified in the block _and_ those in `node_modules`.
